lib/taskana-core/src/main/java/pro/taskana/impl/ClassificationServiceImpl.java

```diff
@@ -221,18 +222,24 @@ private void initDefaultClassificationValues(ClassificationImpl classification)
             try {
                 Duration.parse(classification.getServiceLevel());
             } catch (Exception e) {
-                throw new IllegalArgumentException("Invalid duration. Please use the format defined by ISO 8601");
+                throw new InvalidArgumentException("Invalid duration. Please use the format defined by ISO 8601");
             }
         }
 
         if (classification.getKey() == null) {
-            throw new IllegalStateException("Classification must contain a key");
+            throw new InvalidArgumentException("Classification must contain a key");
         }
 
         if (classification.getParentId() == null) {
             classification.setParentId("");
         }
 
+        if (classification.getType() != null
+            && !taskanaEngine.getConfiguration().getClassificationTypes().contains(classification.getType())) {
```

Comment:
Inkonsistenz im Vergleich zu den Domains (Methode domainExists in der TaskanaEngineImpl)